migrate/014_storage.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    create_table(:vm_storage_volume) do
+      column :id, :uuid, primary_key: true, default: Sequel.lit("gen_random_uuid()")
+      foreign_key :vm_id, :vm, type: :uuid, null: false
+      column :boot, :bool, null: false
+      column :size_gib, :bigint, null: false
+      column :device_id, :text, null: false, unique: true
```

Comment:
Vm names aren't unique globally. They are unique in projects (a.k.a tag spaces).
`device_id: "#{name}0"` will fail when two projects have vm with same name. `name` column is like display name, it's different than vm_name in rhizome.

Reply:
I was also wondering: do we have bdev name clash issues within an SPDK program?

Reply:
sorry, I'm an idiot, it was in the last line of the original message. I guess that's the best we can do without some kind of extra spdk feature for now, though it's unfortunate because it sounds like we'd need to generate a deconflicting name

Reply:
Thanks. I changed the code in this way:

* Instead of having a `device_id text` field, added a `disk_index int` field, and made device_id a method which is computed as `#{uuid_to_name(vm.id)}_#{disk_index}`. We already assume uuid_to_name should return unique names per host, so device_id computation doesn't introduce a new uniqueness problem.
* I moved couple of methods from `Nexus` to `Vm` so they can be called from `VmStorageVolume`.